lib/hexpm_web/controllers/package_report_controller.ex

```diff
@@ -0,0 +1,180 @@
+    all_releases = Releases.all(package)
+
+    report_releases = slice_releases(all_releases, requirement)
+
+    if report_releases == [] do
+      conn
+      |> put_flash(:error, @report_bad_version_msg)
+      |> put_status(400)
+      |> new(%{"package" => package_name})
+    else
+      %{
+        "package" => package,
+        "releases" => report_releases,
+        "user" => user,
+        "description" => description,
+        "state" => state
+      }
+      |> PackageReports.add()
+
+      conn
+      |> redirect(to: Routes.package_report_path(HexpmWeb.Endpoint, :index))
+    end
+  end
+
+  def show(conn, params) do
+    report = PackageReports.get(params["id"])
+    to_moderator = Users.has_role(conn.assigns.current_user, "moderator")
+
+    if report == nil or
+         (report.state == "to_accept" and !to_moderator) do
+      conn
+      |> put_flash(:error, @report_not_accessible)
+      |> put_status(400)
+      |> redirect(to: Routes.package_path(HexpmWeb.Endpoint, :index))
+    else
+      render(
+        conn,
+        "show.html",
+        report: report,
+        to_moderator: to_moderator
+      )
+    end
+  end
+
+  def accept_report(conn, params) do
+    report_id = params["report_id"]
+    comment = params["comment"]
+
+    report = PackageReports.get(report_id)
+
+    if valid_state_change("accepted", report) do
+      PackageReports.accept(conn, report_id, comment)
+      notify_good_update(conn)
+    else
+      notify_bad_update(conn, %{"id" => report_id})
+    end
+  end
+
+  def reject_report(conn, params) do
+    report_id = params["report_id"]
+    comment = params["comment"]
+
+    report = PackageReports.get(report_id)
+
+    if valid_state_change("rejected", report) do
+      PackageReports.reject(conn, report_id, comment)
+      notify_good_update(conn)
+    else
+      notify_bad_update(conn, %{"id" => report_id})
+    end
+  end
+
+  def solve_report(conn, params) do
+    report_id = params["report_id"]
+    comment = params["comment"]
+
+    report = PackageReports.get(report_id)
+
+    if valid_state_change("solved", report) do
+      PackageReports.solve(conn, report_id, comment)
+      notify_good_update(conn)
+    else
+      notify_bad_update(conn, %{"id" => report_id})
+    end
+  end
+
+  defp notify_good_update(conn) do
+    conn
+    |> put_flash(:info, @report_updated_msg)
+    |> redirect(to: Routes.page_path(HexpmWeb.Endpoint, :index))
+  end
+
+  defp notify_bad_update(conn, params) do
+    conn
+    |> put_flash(:error, @report_bad_update_msg)
+    |> put_status(400)
+    |> show(params)
+  end
+
+  defp valid_state_change(state, report) do
+    # TODO: check valid state change based on previous state and conn.user
+    false
+  end
+
+  defp slice_releases(releases, requirement) do
+    rs =
+      Enum.filter(releases, fn r ->
+        Version.match?(r.version, requirement)
+      end)
+
+    rs
+  end
+
+  defp clean_dots(version) do
+    String.replace("#{version}", ".", "")
+  end
+
```

Comment:
This is unused, let's remove it. Please also run `mix compile --warnings-as-errors` and remove any other warnings.

```suggestion
```